apps/root/src/hooks/useAddCustomTokenToList.ts

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { useAppDispatch } from './state';
+import { fetchCustomTokenBalance } from '@state/balances/actions';
+import { Address } from 'viem';
+import { useCustomTokens } from '@state/token-lists/hooks';
+import { getTokenListId } from '@common/utils/parsing';
+import useSelectedNetwork from './useSelectedNetwork';
+import useTrackEvent from './useTrackEvent';
+
+function useAddCustomTokenToList(chainId?: number) {
+  const dispatch = useAppDispatch();
+  const [customTokenAddress, setCustomTokenAddress] = React.useState<Address | undefined>();
+  const [isLoadingCustomToken, setIsLoadingCustomToken] = React.useState(false);
+  const customTokens = useCustomTokens();
+  const selectedNetwork = useSelectedNetwork();
+  const trackEvent = useTrackEvent();
+
+  React.useEffect(() => {
+    const handleCustomTokenBalances = async (tokenAddress: Address) => {
+      const network = chainId || selectedNetwork.chainId;
```

Comment:
selectedNetwork is not on the dependency list

Reply:
Also customTokens is not on the dependencyList, and when both are on the dependency list we'll get a lot more fetching of the custom token due to re-renders caused by not the tokenAddress, so we probably need to check for isLoading or that the addresses are different etc

Reply:
I'm not finding a case where we want to trigger this based on an update from network/customTokens. For our current cases, we are controlling this hook by just triggering it with a new tokenAddress